7.1 Snapshot.md

```diff
@@ -0,0 +1,59 @@
+
+## 7.1.5 Pruning dependencies
+
+Pruning of messages is done when the messages reach the age `DELTA_PRUNE`. 
+
+We require several conditions to be fulfilled in order to allow for a smooth pruning process: 
+- Messages cannot attach to a message whose timestamp has an age difference of more than `DELTA`, see [4.3.4 R-URTS](./4.3%20Tip%20Selection%20Algorithm.md/#4.3.4%20R-URTS).
+- Timestamp precision is enforced through voting, see [4.2.3 General Timestamp rules](./4.2%20Timestamps.md/#4.2.3%20General%20Timestamp%20rules). 
+- Transaction must pass a relaxed past cone check, see [2.4.4.6 Booker - Point 7](./2.4%20Data%20Flow.md/#2.4.4.6%20Booker). 
+- Finalization of messages must be guaranteed to happen before the point of pruning. Thus, any message must be either confirmed accepted or confirmed rejected before the elapse of the pruning age. 
+- The filtering out of messages with wrong timestamps, voting on conflicts and the accumulation of sufficient approval weight takes a time much smaller than `DELTA_PRUNE`. 
+- In order to allow a smooth pruning, any message at the age `DELTA_PRUNE` must either belong to the branch `masterBranch` or to a rejected branch. This simply means that messages are either confirmed or orphaned. 
+
+With the above conditions we can safely assume any message before the time `DELTA_PRUNE>>DELTA` can be pruned. In summary this is for the following reasons:
+- Timestamps are sufficiently accurate.
+- Messages can only create parent-child relationships with messages that have a timestamp that is more recent than a predefined age.
+- Branches are cleaned up at the depth of pruning, i.e. all messages that have pruning age are either confirmed accepted or confirmed rejected.
+- Transactions are only pruned once all of their outputs are confirmed spent. 
+
+
+## 7.1.6 Pruning process
+
+A UTXO output is considered *confirmed spent* , when the message containing a transaction spending it is confirmed accepted. A flag `confirmedSpent` *shall* be set to `TRUE` in the metadata of the output when this event is triggered.
+
+Once a message reaches the age `DELTA_PRUNE` several checks are performed:
+
++ If the message is not a value message it can be removed. 
++ If the message is a value message and is confirmed rejected, i.e. orphaned, it can be pruned. 
++ If the message is a value message and is confirmed accepted (i.e. is confirmed and a member of the `masterBranch`), it is checked whether any of the outputs is not confirmed spent.
++ If all outputs are confirmed spent, then the value message is pruned. 
++ If any of the outputs is unspent, i.e. not confirmed spent, the message *shall* be stored, and its references to parents are set to the `genesis`.
```

Comment:
`and its references to parents are set to the `genesis``
What exactly does that mean and why is it done?

Reply:
Actually this was related to the way i envisioned it to be implemented. but i think for the purpose of this document it just creates confusion and thus i removed it.

Reply:
```suggestion
+ If any of the outputs is unspent, i.e. not confirmed spent, the message *shall* be stored, and kept until all outputs are confirmed spent.
```